miasm2/arch/x86/sem.py

```diff
@@ -2555,7 +2555,12 @@ def fcmovnu(ir, instr, arg1, arg2):
 
 def nop(ir, instr, a=None):
     return [], []
-
+    
+def ud2(ir, instr, a=None):
+    #http://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html
+    #http://blog.regehr.org/archives/213
+    #since we are undefined -> we are fine with nothing
+    return [], []
```

Comment:
The execution of `UD2` raises an `Illegal instruction` in x86. This has to be reflected in the semantic, so you have to do something like:
    e = [m2_expr.ExprAff(exception_flags, m2_expr.ExprInt32(EXCEPT_ILLEGAL_INSN))]

This will trigger the exception in the emulation for example.